src/vs/workbench/contrib/inlineChat/browser/inlineChatDecorations.ts

```diff
@@ -125,14 +125,26 @@ export class InlineChatDecorationsContribution extends Disposable implements IEd
 		decorationUpdateScheduler.schedule();
 	}
 
-	private _onSelectionOrContentChanged(editor: IActiveCodeEditor): void {
+	private async _onSelectionOrContentChanged(editor: IActiveCodeEditor): Promise<void> {
```

Comment:
This code is executed on cursor change (after a delay), so we need to optimize it more. It should avoid doing expensive operations.
- only ask for words if necessary (selection is non-empty)
- remember if there's an inline chat session open or not.
- maybe remember the breakpoints for the current URI

Reply:
I just saw that this method did not need to be async, it can be synchronous. The async part is legacy code from when I was working on it before. In this case do we still want to ask for the words when necessary?